### PR TITLE
(Closes #1903) rationalise OMP schedule values

### LINKED ISSUE #1903: Rationalise OpenMP default schedule
In my previous OpenMP PR #1853 I changed the default schedule from 'static' to 'auto'. As it seemed a better default, but I was wrong. 

OpenMP default schedule is RUNTIME, and if the OMP_RUNTIME is not found then the default is implementation-dependent (but usually implemented as STATIC).

Currently for offloading with nvidia compiler the default produces a compiler error:
`NVFORTRAN-F-1196-OpenMP - Only static schedule kind is supported for GPU`
if I ask for "runtime" it also says:
`NVFORTRAN-F-1196-OpenMP - Only static schedule kind is supported for GPU`
but changing it to static produces a runtime error:
`'HX_CU_CALL_CHECK(p_cuStreamSynchronize(stream[dev]))' (value 1) is not equal to expression 'HX_SUCCESS' (value 0)`

The only working option in tra-adv is to not specify the schedule clause.

This could seem a particular issue with the nvidia implementation but there is another reason in the standard to support not specifying any schedule. In collapse loops it says: `The schedule clause cannot be specified if any of the associated loops are non-rectangular.`

What I think PSyclone should be doing is adding a new option to the valid schedules. (maybe empty string or special value?)
VALID_OMP_SCHEDULES = ["runtime", "static", "dynamic", "guided", "auto", ""]

That supports not adding a schedule clause at all in the directive statement, and this be the default.

### PR DESCRIPTION
**Pull request by arporter**
Changes the default OMP schedule to be "none" which means that no `schedule` clause is added to the OpenMP directive. Also refactors things slightly so that the `OMPScheduleClause` is the single source of truth for permitted values of schedule.

### CODE REVIEW

**Comment by arporter** [comment]
Note to self: check testing of OMPParallelDoDirective.gen_code as test for `schedule` rather than `schedule != "none" wasn't picked up.

**Comment by arporter** [comment]
I need to check the various examples that involve OpenMP and then this can go for review.

**Comment by arporter** [comment]
I've checked, and the various examples that show OpenMP appear to be correct.

Ready for first review from @sergisiso.

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter This is a nice improvement on the OpenMP schedule behavior. There is just some minor suggestions to fix. pycodestyle, pylint, tests, codecov, TODOs and documentation building is all good.

A further question is now that the behavior matches the OpenMP standard would it make sense to remove the LFRic and Gocean own default to 'static' schedule or do you prefer to keep them.
(in practice it doesn't change much because most implementations default to static if no env var is found - it is mainly to be closer to the standard)

File: src/psyclone/tests/psyir/nodes/omp_clause_test.py [hunk 1]
```diff
@@ -102,8 +105,8 @@ def test_invalid_schedule_clause():
     sched = OMPScheduleClause()
     with pytest.raises(ValueError) as excinfo:
         sched.schedule = "test"
-    assert ("Schedule must be one of runtime, static, dynamic, guided or auto."
-            " Found test." in str(excinfo.value))
+    assert ("Schedule must be one of ['runtime', 'static', 'dynamic', "
+            "'guided', 'auto', 'none']. Found test." in str(excinfo.value))
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/omp_clause_test.py:109]: Oops.

File: src/psyclone/tests/psyir/nodes/omp_directives_test.py [hunk 2]
```diff
@@ -153,7 +153,8 @@ def test_ompparallel_changes_gen_code():
 
 def test_omp_paraleldo_changes_gen_code():
     ''' Check that when the code inside an OMP Parallel Do region changes, the
-    private clause changes appropriately. '''
+    private clause changes appropriately. Also check that changing the schedule
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/omp_directives_test.py:156]: Good spot!

File: src/psyclone/tests/psyir/nodes/omp_directives_test.py [hunk 3]
```diff
@@ -170,21 +171,24 @@ def test_omp_paraleldo_changes_gen_code():
...
+    # Make a copy of the loop
     routine = pdir.ancestor(Routine)
     routine.symbol_table.add(DataSymbol("k", INTEGER_SINGLE_TYPE))
     # Change the loop variable to j
     jvar = DataSymbol("k", INTEGER_SINGLE_TYPE)
     pdir.children[0].children[0].variable = jvar
-    # Change the schedule
-    pdir._omp_schedule = "dynamic"
+    # Change the schedule to 'none'
+    pdir._omp_schedule = "none"
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/omp_directives_test.py:181]: Oh yes.

File: src/psyclone/tests/psyir/nodes/omp_directives_test.py [hunk 4]
```diff
@@ -170,21 +171,24 @@ def test_omp_paraleldo_changes_gen_code():
...
 
-    _ = psy.gen
+    code = str(psy.gen).lower()
     assert pdir.children[2] != priv_clause
     assert isinstance(pdir.children[2], OMPPrivateClause)
     assert pdir.children[3] != sched_clause
     assert isinstance(pdir.children[3], OMPScheduleClause)
 
+    # No 'schedule' clause should now be present on the OMP directive.
+    assert "schedule(" not in code
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/omp_directives_test.py:190]: Done.

**Comment by arporter** [comment]
CI permitting, this is ready for another look. I'd prefer to leave the GOcean and LFRic defaults alone for now.

**Review by sergisiso — APPROVED** [review]
@arporter All comments addressed and all check/formatting/docs still pass. This is approved to merge.

**Pull request merged**